ladybug_geometry/geometry2d/pointvector.py

```diff
@@ -0,0 +1,337 @@
+# coding=utf-8
+"""2D Vector and 2D Point"""
+from __future__ import division
+
+import math
+import operator
+
+
+class Vector2D(object):
+    """2D Vector object.
+
+    Properties:
+        x
+        y
+        magnitude
+        magnitude_squared
+    """
+    __slots__ = ('_x', '_y')
+
+    def __init__(self, x=0, y=0):
+        """Initialize 2D Vector."""
+        self._x = x
+        self._y = y
+
+    @property
+    def x(self):
+        """Get the X coordinate."""
+        return self._x
+
+    @property
+    def y(self):
+        """Get the Y coordinate."""
+        return self._y
+
+    @property
+    def magnitude(self):
+        """Get the magnitude of the vector."""
+        return self.__abs__()
+
+    @property
+    def magnitude_squared(self):
+        """Get the magnitude squared of the vector."""
+        return self.x ** 2 + self.y ** 2
+
+    def normalize(self):
+        """Get a copy of the vector that is a unit vector (magnitude=1)."""
+        d = self.magnitude
+        if d:
+            return Vector2D(self.x / d, self.y / d)
+        return self.duplicate()
```

Comment:
This will only be the case if the vector is (0, 0). Right? Why do we need to treat this situation differently?

Reply:
We want to avoid an uncaught `ZeroDivisionError`.  I guess I could just do a `try / except ZeroDivisionError` instead of an `if` statement.  Does that sounds preferable?

Reply:
After thinking about this more, I think the try / except is better. I will implement it.